src/main/java/edu/ie3/datamodel/io/sink/InfluxDbSink.java

```diff
@@ -0,0 +1,311 @@
+    this.connector = connector;
+    this.fileNamingStrategy = fileNamingStrategy;
+    this.processorProvider =
+        new ProcessorProvider(
+            ProcessorProvider.allResultEntityProcessors(),
+            ProcessorProvider.allTimeSeriesProcessors());
+  }
+
+  /**
+   * Initializes a new InfluxDbWeatherSource with a default FileNamingStrategy
+   *
+   * @param connector needed for database connection
+   */
+  public InfluxDbSink(InfluxDbConnector connector) {
+    this(connector, new FileNamingStrategy());
+  }
+
+  @Override
+  public void shutdown() {
+    connector.shutdown();
+  }
+
+  @Override
+  public <C extends UniqueEntity> void persist(C entity) {
+    Set<Point> points = extractPoints(entity);
+    writeAll(points);
+  }
+
+  @Override
+  public <C extends UniqueEntity> void persistAll(Collection<C> entities) {
+    Set<Point> points = new HashSet<>();
+    for (C entity : entities) {
+      points.addAll(extractPoints(entity));
+    }
+    writeAll(points);
+  }
+
+  @Override
+  public <C extends UniqueEntity> void persistIgnoreNested(C entity) {
+    try {
+      if (!(entity instanceof ResultEntity))
+        throw new SinkException("Cannot handle entities that are not ResultEntities");
+      Optional<Point> influxPoint = transformToPoint((ResultEntity) entity);
+      write(influxPoint.orElseThrow(() -> new SinkException("Could not transform entity")));
+    } catch (SinkException e) {
+      log.error(
+          "Cannot persist provided entity '{}'. Exception: {}",
+          () -> entity.getClass().getSimpleName(),
+          () -> e);
+    }
+  }
+
+  @Override
+  public <C extends UniqueEntity> void persistAllIgnoreNested(Collection<C> entities) {
+    Set<Point> points = new HashSet<>();
+    /* Distinguish between result models and time series */
+    for (C entity : entities) {
+      try {
+        if (!(entity instanceof ResultEntity))
+          throw new SinkException("Cannot handle entities that are not ResultEntities");
+        points.add(
+            transformToPoint((ResultEntity) entity)
+                .orElseThrow(() -> new SinkException("Could not transform entity")));
+      } catch (SinkException e) {
+        log.error(
+            "Cannot persist provided entity '{}'. Exception: {}",
+            () -> entity.getClass().getSimpleName(),
+            () -> e);
+      }
+    }
+    writeAll(points);
+  }
+
+  @Override
+  public <E extends TimeSeriesEntry<V>, V extends Value> void persistTimeSeries(
+      TimeSeries<E, V> timeSeries) {
+    Set<Point> points = transformToPoints(timeSeries);
+    writeAll(points);
+  }
+
+  @Override
+  public void persistJointGrid(JointGridContainer jointGridContainer) {
+    log.error("Cannot persist grids");
+  }
+
+  /**
+   * Transforms a ResultEntity to an influxDB data point. <br>
+   * The measurement point will be named by the given FileNamingStrategy if possible, or the class
```

Comment:
What is _measurement point_ for you? I find this naming is somehow not expressive. At least I cannot figure out what this should be.

Reply:
Ah okay, I see. This is some InfluxDB term. Maybe you can make this a little clearer. In our model we also have models of physical measurement devices. I think, those can easily be mixed up.